### PR TITLE
Improve AWSCluster docs and changelog

### PR DESCRIPTION
**Pull request by paurosello**
## Checklist

- [x] Consider SIG UX feedback.
- [x] Update changelog in CHANGELOG.md.
- [x] If adding a new type, ensure that it is added to the [CRD unit tests](https://github.com/giantswarm/apiextensions/blob/d78aba91d578d56ef49f58e336035d35edc4e1b0/pkg/crd/crd_test.go#L9).

### CODE REVIEW

**Comment by paurosello** [comment]
We can merge anytime, we don't need a new operator release for this, maybe the case for opsctl so this CRD is what you can see in the Control Planes

**Comment by marians** [comment]
I'd like to push a change or two into this branch.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,14 +7,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 
 ## [Unreleased]
-
+- Improve documentation
```
Comment (kopiczko) [inline CHANGELOG.md:10]: I'd rather skip that.

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,14 +7,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 
 
 ## [0.3.0] - 2020-04-16
 
 ### Changed
 
-- Add pod field to AWSCluster.
+- Add `.spec.provider.pods` field to AWSCluster.
```
Comment (kopiczko) [inline CHANGELOG.md:17]: This should go under `### Added` section.

**Comment by marians** [comment]
@paurosello I'd appreciate if we could wrap this up :)

**Comment by tfussell** [comment]
Still working on this? And if so, need a hand with the merge conflicts?

**Comment by marians** [comment]
Let me answer for team firecracker: That would be perfect! ❤️

**Comment by tfussell** [comment]
> Let me answer for team firecracker: That would be perfect! ❤️

Okay, I'll try to work it in tomorrow or Wednesday.

**Comment by MarcelMue** [comment]
Closing in favour of:
https://github.com/giantswarm/apiextensions/pull/466

**Pull request closed without merging**